eventmesh-server-go/plugin/registry/nacos/nacos.go

```diff
@@ -0,0 +1,241 @@
+	DefaultClusterName            = "DEFAULT"
+	DefaultGroupName              = "DEFAULT_GROUP"
+	DefaultWeight         float64 = 10
+	protoList                     = []string{"TCP", "HTTP", "GRPC"}
+)
+
+func init() {
+	plugin.Register("nacos", &Registry{})
+}
+
+type Registry struct {
+	initStatus    *atomic.Bool
+	startStatus   *atomic.Bool
+	cfg           *Config
+	client        naming_client.INamingClient
+	registryInfos *sync.Map // map[string]*registry.EventMeshRegisterInfo
+}
+
+func (r *Registry) Type() string {
+	return registry.Type
+}
+
+func (r *Registry) Setup(name string, dec plugin.Decoder) error {
+	conf := &Config{}
+	if err := dec.Decode(conf); err != nil {
+		return err
+	}
+	r.cfg = conf
+	return nil
+}
+
+func (r *Registry) Init() error {
+	r.initStatus = atomic.NewBool(true)
+	r.startStatus = atomic.NewBool(false)
+	r.registryInfos = new(sync.Map)
+	return nil
+}
+
+func (r *Registry) Start() error {
+	var p vo.NacosClientParam
+	var addresses []string
+
+	if len(r.cfg.AddressList) > 0 {
+		addresses = strings.Split(r.cfg.AddressList, ",")
+	}
+
+	for _, address := range addresses {
+		ip, port, err := net.SplitHostPort(address)
+		if err != nil {
+			return err
+		}
+		p.ServerConfigs = append(p.ServerConfigs, constant.ServerConfig{IpAddr: ip, Port: gconv.Uint64(port)})
+	}
+
+	p.ClientConfig = &constant.ClientConfig{
+		TimeoutMs: defaultConnectTimeout,
+		CacheDir:  r.cfg.CacheDir,
+	}
+	cli, err := clients.NewNamingClient(p)
+	if err != nil {
+		return err
+	}
+
+	r.startStatus.Store(true)
+	r.client = cli
+	return nil
+}
+
+func (r *Registry) Shutdown() error {
+	r.startStatus.Store(false)
+	r.initStatus.Store(false)
+	r.client.CloseClient()
+	return nil
+}
+
+func (r *Registry) FindEventMeshInfoByCluster(clusterName string) ([]*registry.EventMeshDataInfo, error) {
+	var infos []*registry.EventMeshDataInfo
+	meshServerName := config.GlobalConfig().Common.Name
+	cluster := config.GlobalConfig().Common.Cluster
+	for _, proto := range protoList {
+		ins, err := r.client.SelectInstances(vo.SelectInstancesParam{
+			Clusters:    []string{clusterName},
+			ServiceName: fmt.Sprintf("%v-%v", meshServerName, proto),
+			GroupName:   cluster,
+			HealthyOnly: true,
+		})
+		if err != nil {
+			return nil, err
+		}
+		if len(ins) == 0 {
+			continue
+		}
+		for _, in := range ins {
+			infos = append(infos, &registry.EventMeshDataInfo{
+				EventMeshClusterName: in.ClusterName,
+				EventMeshName:        in.ServiceName,
+				Endpoint:             fmt.Sprintf("%v:%v", in.Ip, in.Port),
+				LastUpdateTimestamp:  time.Time{},
+				Metadata:             in.Metadata,
+			})
+		}
+	}
+
+	return infos, nil
+}
+
+func (r *Registry) FindAllEventMeshInfo() ([]*registry.EventMeshDataInfo, error) {
+	var infos []*registry.EventMeshDataInfo
+	meshServerName := config.GlobalConfig().Common.Name
+
+	for _, proto := range protoList {
+		ins, err := r.client.SelectInstances(vo.SelectInstancesParam{
+			Clusters:    []string{},
```

Comment:
Why not use wildcard?

Reply:
I write it with reference to the implementation of java:
https://github.com/apache/incubator-eventmesh/blob/master/eventmesh-registry-plugin/eventmesh-registry-nacos/src/main/java/org/apache/eventmesh/registry/nacos/service/NacosRegistryService.java#L165

@xwm1992 Can this be changed to `SelectAllInstances`.

Reply:
`SelectAllInstances` will return all instances, but `SelectInstances` only return the instances of healthy
```
	// SelectAllInstances return all instances,include healthy=false,enable=false,weight<=0
	// ServiceName require
	// Clusters optional,default:DEFAULT
	// GroupName optional,default:DEFAULT_GROUP
	SelectAllInstances(param vo.SelectAllInstancesParam) ([]model.Instance, error)

	// SelectInstances only return the instances of healthy=${HealthyOnly},enable=true and weight>0
	// ServiceName require
	// Clusters optional,default:DEFAULT
	// GroupName optional,default:DEFAULT_GROUP
	// HealthyOnly optional
	SelectInstances(param vo.SelectInstancesParam) ([]model.Instance, error)

```